src/components/textField/textField.api.json

```diff
@@ -64,6 +64,7 @@
     },
     {"name": "validateOnStart", "type": "boolean", "description": "Should validate when the TextField mounts"},
     {"name": "validateOnChange", "type": "boolean", "description": "Should validate when the TextField value changes"},
+    {"name": "timeoutOnChange", "type": "number", "description": "Add a debounce timeout when sending validateOnChange"},
```

Comment:
the name is misleading it sounds like it will debounce the onChange callback when typing 
I would rename to `validationDebounceTime`